Filled in details for multi-select element (#2330)
reactjs/reactjs.org: content/docs/forms.md

Overall, this makes it so that `<input type="text">`, `<textarea>`, and `<select>` all work very similarly - they all accept a `value` attribute that you can use to implement a controlled component.

### The select multiple Tag

> Just like in HTML, select elements can become multi-select elements through the addition of the *multiple* attribute to the select tag. But handling of the value attribute in react is more complex due to having an array of values. 

Reviewer: ```suggestion
Just like in HTML, select elements can become multi-select elements through the addition of the *multiple* attribute to the select tag. In React, the `value` attribute for `multiple` is handled as an array. 
```

Author: I see where you're going with that but I don't like your wording of it either. How about: 
> In React, this results in the `value` attribute being handled as an array of selected values, rather than just one value.

Here's our example select element with multi-select support:

```js{4,11,24}
class FlavorForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {value: ['coconut']};

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(event) {
    this.setState({value: Array.from(event.target.options).filter(o => o.selected).map(o => o.value)});
  }

  handleSubmit(event) {

>     alert('Your favorite flavor is: ' + this.state.value);

Reviewer: It might make a bit more sense if this used a more multiple + plural example, like "Your favorite flavors are" or "You picked: ..."

Author: I thought about that, but then I thought if this is my favorite flavor of ice cream then it would be multiple flavors. So the language actually makes sense in the current context.
